Lib/test/test_zipfile/test_core.py

```diff
@@ -3447,6 +3447,62 @@ def test_too_short(self):
         self.assertEqual(
             b"zzz", zipfile._Extra.strip(b"zzz", (self.ZIP64_EXTRA,)))
 
+class StoredZipExtFileRandomReadTest(unittest.TestCase):
+    def test_random_read(self):
+        from _pyio import BytesIO
+        class StatIO(BytesIO):
+            def __init__(self):
+                super().__init__()
+                self.bytes_read = 0
+
+            def read(self, size=-1):
+                bs = super().read(size)
+                self.bytes_read += len(bs)
+                return bs
+
+        sio = StatIO()
+        # 20000 bytes
+        txt = b'0123456789' * 2000
+
+        # seek length must be greater than ZipExtFile.MIN_READ_SIZE (4096)
```

Comment:
```suggestion
        # seek length must be greater than ZipExtFile.MIN_READ_SIZE (4096)
        # as `ZipExtFile._read2()` reads in blocks of this size and we need to
        # seek out of the buffered data
```

Reply:
Sorry, that comment wasn't very good. This one is better?

```suggestion
        # seek length must be greater than ZipExtFile.MIN_READ_SIZE (4096)
        # as `ZipExtFile._read2()` reads in blocks of this size and this test relies
        # on reads remaining within one of these blocks.
```

Reply:
Just to clarify, it was my suggested change that was not very good. It didn’t describe what was happening in the code as well as it could :)

Reply:
I think the first one is better